### PR TITLE
Improve `MyPy` typing for `core.cli`

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
- Enable `mypy` check on `core.cli`.
- Correct some typos.
- Update some test according to the new typing.

### CODE REVIEW

File: parsec/core/cli/rsync.py [hunk 1]
```diff
@@ -96,21 +106,25 @@ async def _clear_directory(
...
+        manifest = await workspace_fs.remote_loader.load_manifest(entry_id)
+        assert manifest is None or isinstance(
+            manifest, FolderManifest
+        ), f"Not a folder manifest: {type(manifest).__name__}"
+        folder_manifest: FolderManifest | None = manifest
+        return folder_manifest
     else:
         folder_manifest = await _create_path(workspace_fs, True, local_path, workspace_path)
-    return folder_manifest
+        return folder_manifest
```
Comment (TimeEngineer) [inline parsec/core/cli/rsync.py:121]: why 2nd return ?

File: parsec/core/cli/rsync.py [hunk 2]
```diff
@@ -159,39 +174,50 @@ def _parse_destination(
...
 
-    root_manifest = workspace_manifest
-    parent = FsPath("/")
+    root_manifest: WorkspaceManifest | FolderManifest = workspace_manifest
+    # parent = FsPath("/")
+    parent = trio.Path("/")
     if path_destination:
         for p in path_destination.parts:
-            parent = FsPath(parent / p)
+            # parent = FsPath(parent / p)
```
Comment (TimeEngineer) [inline parsec/core/cli/rsync.py:200]: Why comment ?

File: parsec/core/cli/rsync.py [hunk 3]
```diff
@@ -159,39 +174,50 @@ def _parse_destination(
...
 
 
 async def _root_manifest_parent(
     path_destination: FsPath, workspace_fs: WorkspaceFS, workspace_manifest: WorkspaceManifest
 ) -> Tuple[Union[WorkspaceManifest, FolderManifest], FsPath]:
 
-    root_manifest = workspace_manifest
-    parent = FsPath("/")
+    root_manifest: WorkspaceManifest | FolderManifest = workspace_manifest
+    # parent = FsPath("/")
```
Comment (TimeEngineer) [inline parsec/core/cli/rsync.py:196]: Why comment ?

File: parsec/core/cli/utils.py [hunk 4]
```diff
@@ -30,10 +30,12 @@
...
 
 
-F = TypeVar("F", bound=Callable)
+R_gui_command_base_options = TypeVar("R_gui_command_base_options")
 
 
-def gui_command_base_options(fn: F) -> F:
+def gui_command_base_options(
+    fn: Callable[..., R_gui_command_base_options]
+) -> Callable[..., R_gui_command_base_options]:
```
Comment (TimeEngineer) [inline parsec/core/cli/utils.py:38]: You can use ParamSpec to forward `type`

File: parsec/core/cli/utils.py [hunk 5]
```diff
@@ -48,7 +50,12 @@ def gui_command_base_options(fn: F) -> F:
     return fn
 
 
-def cli_command_base_options(fn: F) -> F:
+R_cli_command_base_options = TypeVar("R_cli_command_base_options")
+
+
+def cli_command_base_options(
+    fn: Callable[..., R_cli_command_base_options]
+) -> Callable[..., R_cli_command_base_options]:
```
Comment (TimeEngineer) [inline parsec/core/cli/utils.py:58]: Same

File: parsec/core/cli/utils.py [hunk 6]
```diff
@@ -63,7 +70,10 @@ def cli_command_base_options(fn: F) -> F:
     return fn
 
 
-def core_config_options(fn: F) -> F:
+R_core_opts = TypeVar("R_core_opts")
+
+
+def core_config_options(fn: Callable[..., R_core_opts]) -> Callable[..., R_core_opts]:
```
Comment (TimeEngineer) [inline parsec/core/cli/utils.py:76]: Same

File: parsec/core/cli/utils.py [hunk 7]
```diff
@@ -114,7 +127,12 @@ def format_available_devices(devices: List[AvailableDevice]) -> str:
     return "\n".join(out)
 
 
-def core_config_and_available_device_options(fn: F) -> F:
+R_core_config_and_available_device_options = TypeVar("R_core_config_and_available_device_options")
+
+
+def core_config_and_available_device_options(
+    fn: Callable[..., R_core_config_and_available_device_options]
+) -> Callable[..., R_core_config_and_available_device_options]:
```
Comment (TimeEngineer) [inline parsec/core/cli/utils.py:135]: Same

File: parsec/core/cli/utils.py [hunk 8]
```diff
@@ -162,7 +185,12 @@ def wrapper(**kwargs):
     return wrapper
 
 
-def core_config_and_device_options(fn: F) -> F:
+R_core_config_and_device_options = TypeVar("R_core_config_and_device_options")
+
+
+def core_config_and_device_options(
+    fn: Callable[..., R_core_config_and_device_options]
+) -> Callable[..., R_core_config_and_device_options]:
```
Comment (TimeEngineer) [inline parsec/core/cli/utils.py:193]: Same

File: parsec/core/cli/utils.py [hunk 9]
```diff
@@ -195,12 +228,17 @@ def wrapper(**kwargs):
     return wrapper
 
 
-def save_device_options(fn: F) -> F:
+R_save_device_options = TypeVar("R_save_device_options")
+
+
+def save_device_options(
+    fn: Callable[..., R_save_device_options]
+) -> Callable[..., R_save_device_options]:
```
Comment (TimeEngineer) [inline parsec/core/cli/utils.py:236]: Same

File: parsec/core/cli/bootstrap_organization.py [hunk 10]
```diff
@@ -55,27 +63,25 @@ async def _bootstrap_organization(
         )
         if not answer:
             raise SystemExit("Bootstrap aborted")
-        sequester_verify_key = SequesterVerifyKeyDer(sequester_verify_key.read_bytes())
+        sequester_vrf_key = SequesterVerifyKeyDer(sequester_verify_key.read_bytes())
```
Comment (TimeEngineer) [inline parsec/core/cli/bootstrap_organization.py:66]: Why rename ?

File: parsec/core/cli/bootstrap_organization.py [hunk 11]
```diff
@@ -55,27 +63,25 @@ async def _bootstrap_organization(
...
-    if not human_email:
-        human_email = await aprompt("User email")
+    human_label: str = human_label or await aprompt("User fullname")
+    human_email: str = human_email or await aprompt("User email")
     human_handle = HumanHandle(email=human_email, label=human_label)
     if not device_label:
-        device_label_raw = await aprompt("Device label", default=platform.node())
-        device_label = DeviceLabel(device_label_raw)
+        device_label_raw: str = await aprompt("Device label", default=platform.node())
+        dev_label = DeviceLabel(device_label_raw)
```
Comment (TimeEngineer) [inline parsec/core/cli/bootstrap_organization.py:73]: Why rename ?

**Review by TimeEngineer — CHANGES_REQUESTED** [review]

File: parsec/core/cli/utils.py (hunk 4 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/utils.py:38]: Nop, too complex I've already tried it

File: parsec/core/cli/utils.py (hunk 5 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/utils.py:58]: see https://github.com/Scille/parsec-cloud/pull/3320#discussion_r998307433

File: parsec/core/cli/utils.py (hunk 6 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/utils.py:76]: see https://github.com/Scille/parsec-cloud/pull/3320#discussion_r998307433

File: parsec/core/cli/utils.py (hunk 7 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/utils.py:135]: see https://github.com/Scille/parsec-cloud/pull/3320#discussion_r998307433

File: parsec/core/cli/utils.py (hunk 8 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/utils.py:193]: see https://github.com/Scille/parsec-cloud/pull/3320#discussion_r998307433

File: parsec/core/cli/utils.py (hunk 9 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/utils.py:236]: see https://github.com/Scille/parsec-cloud/pull/3320#discussion_r998307433

File: parsec/core/cli/bootstrap_organization.py (hunk 11 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/bootstrap_organization.py:73]: Same as above, `mypy` don't like to pass from a `str` to a `DeviceLabel` for the same variable

File: parsec/core/cli/rsync.py (hunk 2 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/rsync.py:200]: it's an omission

Comment (FirelightFlagboy) [reply parsec/core/cli/rsync.py:200]: done

File: parsec/core/cli/rsync.py (hunk 3 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/rsync.py:196]: done

File: parsec/core/cli/rsync.py (hunk 1 above)
Comment (FirelightFlagboy) [reply parsec/core/cli/rsync.py]: Yes, I can remove one now

Comment (FirelightFlagboy) [reply parsec/core/cli/rsync.py]: done

**Review by TimeEngineer — APPROVED** [review]

**Review by Ironicbay — APPROVED** [review]

File: tests/test_cli.py [hunk 12]
```diff
@@ -739,7 +739,7 @@ def test_full_run(coolorg, unused_tcp_port, tmp_path, ssl_conf):
 @pytest.mark.parametrize(
     "env",
     [
-        pytest.param({}, id="Standard environement"),
+        pytest.param({}, id="Standard environnement"),
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply tests/test_cli.py:742]:
```suggestion
        pytest.param({}, id="Standard environment"),
```

[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply tests/test_cli.py]: Done ^^

**Pull request merged**